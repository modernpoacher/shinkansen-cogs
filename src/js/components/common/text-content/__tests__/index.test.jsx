import React from 'react'
import snapshotOf from 'react-component-snapshot'
import renderer from 'react-test-renderer'

import '@testing-library/jest-dom'

import {
  render
} from '@testing-library/react'

import TextContent from '#cogs/components/common/text-content'

describe('#cogs/components/common/text-content', () => {
  describe('<TextContent />', () => {
    it('renders', () => {
      const {
        container: {
          firstElementChild: textContent
        }
      } = render(
        <TextContent
          textContent='MOCK TEXT CONTENT'
        />
      )

      expect(textContent)
        .toBeInstanceOf(HTMLSpanElement)
    })

    it('matches the snapshot', () => {
      const {
        container: {
          firstElementChild: textContent
        }
      } = render(
        <TextContent
          textContent='MOCK TEXT CONTENT'
        />
      )

      expect(snapshotOf(textContent))
        .toMatchSnapshot()
    })

    /**
     *  @deprecated For migration toward Testing Library
     */
    xit('matches the snapshot', () => {
      expect(renderer.create(<TextContent textContent='MOCK TEXT CONTENT' />).toJSON())
        .toMatchSnapshot()
    })
  })
})
