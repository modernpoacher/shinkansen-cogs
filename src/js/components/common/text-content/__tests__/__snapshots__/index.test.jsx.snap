// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#cogs/components/common/text-content <TextContent /> matches the snapshot 1`] = `
<span
  className="text-content"
>
  MOCK TEXT CONTENT
</span>
`;
